src/hooks/useTheme/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+
+function useTheme() {
+  const [theme, setTheme] = useState<string>(
+    localStorage.getItem("theme") ?? "light"
+  );
+
+  const handleTheme = () => {
+    setTheme(theme === "light" ? "dark" : "light");
+  };
+
+  useEffect(() => {
+    if (theme === "dark") {
+      document.documentElement.classList.add("dark");
```

Comment:
Entiendo que esto no se debería hacer, solo hay que contemplar en los componentes la parte dark: como figura en la documentación de Tailwind y lo único que habría que cambiar es el seteo en el local storage y el cambio (si no recuerdo mal) de tema, sobre el tag de html 

https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually

Reply:
Esto no lo modifique todavía porque no me quedo del todo claro. Lo implementé de esa manera porque el darkmode se activa cuando el HTML o el body tienen la clase "dark" y del mismo modo si no la tienen significa que los componentes van a usar los estilos que no estan declarados como "dark:estilo".

Reply:
yo creo que habria que dar soporte tambien para que respete las preferencia de  usuario como esta aqui en la documentacion [Supporting system preference and manual selection](https://tailwindcss.com/docs/dark-mode#supporting-system-preference-and-manual-selection)